main.tf

```diff
@@ -47,4 +47,7 @@ resource "google_compute_instance" "vm_instance" {
   }
   allow_stopping_for_update = var.allow_stopping_for_update
   depends_on                = [google_project_service.compute_api]
+  lifecycle {
+    ignore_changes = [attached_disk]
+  }
```

Comment:
You may need to rebase from latest master as i have added some lifecycle ignore rules last night

Reply:
sure

Reply:
Just adding `lifecycle {}` block wouldn't make it ignore. Need to add an "initial" value so that Terraform detects any different value there as a "changed" value and not as a newly "inserted" value.

You can refer to [this commit](https://github.com/airasia/terraform-google-vm_instance/commit/fe31d1ec3069cc88fc236737d38bceff7226767b). There is a helping see-reference included in there for more info.